### DISCUSSION TITLE
Can't get automerge to wrok

### DISCUSSION THREAD

**Discussion by garronej** [renovatebot/renovate#16729] [Request Help]
Hi,  

Thank you for this great tool, unfortunately I can get the automerge to work.  
I made [a test repo](https://github.com/garronej/furry-pancake) to try sort this out but now I am out of ideas.  
This is a PR that I think should have been auto merged:  

https://github.com/garronej/furry-pancake/pull/7  

<img width="1309" alt="image" src="https://user-images.githubusercontent.com/6702424/180628368-769381f8-a33a-4a3a-afb0-cd5c39b5f673.png">

The debug log gives me `DEBUG: PR is not ready for merge (branch status is yellow)(branch="renovate_main-garronej_modules_update")`  

Thank you for your help

**Comment by rarkins** [comment]
Has Renovate run again since the tests went green? The debug logs will include more details about why the branch checks are considered yellow, probably right above the log you pasted

**Comment by PhilipAbed** [comment]
@[renovate](https://github.com/apps/renovate) renovate bot merged commit [1ef6d52](https://github.com/garronej/furry-pancake/commit/1ef6d527e6264189001f7f8e7862fc872c021d6c) into main [4 hours ago](https://github.com/garronej/furry-pancake/pull/7#event-7050878819)
1 check passed

i think this is not relevant anymore :D

**Comment by rarkins** [comment]
Set platformAutomerge=true in your config if you want to use GitHub's native automerge on green. Otherwise you need to wait until Renovate runs again on a repo before it can merge.

**Reply by viceice** [reply]
but make sure to have required status checks, otherwise GitHub will merge early and don't wait for checks to pass

**Comment by garronej** [comment]
Thank you rarkins, viceice and PhilipAbed for the quick support, it's much appreciated.  
Okay so the explanation makes sense. I tried to rely on GitHub's automerge feature but now I am facing another problem.  
Whenever the PR is merged as soon as the test complete but... it merges even if the check fails. 😕  

[The repo](https://github.com/etalab/sill-web)  

[My renovate config](https://github.com/etalab/sill-web/blob/ce6d7f14c11177c5bcbdeb9fdeefbbed8548979b/renovate.json#L48-L50)  

[The renovate PR](https://github.com/etalab/sill-web/pull/61)

<img width="811" alt="image" src="https://user-images.githubusercontent.com/6702424/180682065-b2aeade3-6775-4fb4-98b3-10eb97d9db3f.png">

<img width="801" alt="image" src="https://user-images.githubusercontent.com/6702424/180682105-fddd3415-f4d2-48b5-9749-875e99a8e590.png">

<img width="1631" alt="image" src="https://user-images.githubusercontent.com/6702424/180682161-ff2434ef-ed05-48bb-852e-0d594e73aef4.png">

![image](https://user-images.githubusercontent.com/6702424/180682188-e9c70ef9-d39c-4819-97ee-8dc077c69afe.png)

<img width="937" alt="image" src="https://user-images.githubusercontent.com/6702424/180682536-0f9a5f78-cbaf-44b8-9626-479413b47e92.png">
  
Maybe it's on github side, or maybe I haven't understood how it's supposed to work... but I'm stuck again... 😕   

Thank you in advance,

**Reply by viceice** [reply]
https://github.com/renovatebot/renovate/discussions/16729#discussioncomment-3214988

☝️

**Reply by garronej** [reply]
Hi @viceice,  

I saw your reply, I do have setup required status checks, this is what I'm showing in the screenshots.  
I have `Check if version upgrade` that is required.  
<img width="693" alt="image" src="https://user-images.githubusercontent.com/6702424/180723360-2f3cfac2-e725-48d6-9b4c-f0eab11607eb.png">
It does waits for the status 'Check if version upgrade' but it merges even if it's reported skipped because required jobs have failed.   
<img width="809" alt="image" src="https://user-images.githubusercontent.com/6702424/180723634-dc713080-34e7-45e4-9544-b8382d039539.png">

**Reply by rarkins** [reply]
This is a GitHub configuration problem or bug and nothing to do with renovate. If you've set branch protections on GitHub and GitHub failed to protect your branch then you need to raise it with GitHub

**Reply by garronej** [reply]
Ok thanks for clearing that out for me!

**Reply by viceice** [reply]
you need to make build mandatory too

**Reply by garronej** [reply]
Thank you! Indeed, I need to make required all jobs that needs to pass, not only the last one. GitHub see skipped jobs as green, makes sense now that I think about it.  
Thank you for the support and sorry for bothering you with questions unrelated to renovate.  

Renovate will be saving me at least 4 hours a week of chore, much love ❤️
